### PR TITLE
Check for chunk size minimum before using multipart upload with S3

### PR DESCRIPTION
**Pull request by ian-ack-db**
### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

S3 requires chunk sizes to be at least 5 MiB for multipart upload [documentation](https://docs.aws.amazon.com/en_us/AmazonS3/latest/userguide/qfacts.html). This change does an assertion on chunk size env variable before commencing uploads.
### How is this PR tested?

- [ ] Existing unit/integration tests
- [X] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [X] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [X] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [X] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [X] Yes (this PR will be cherry-picked and included in the next patch release)
- [ ] No (this PR will be included in the next minor release)

### CODE REVIEW

File: mlflow/store/artifact/cloud_artifact_repo.py [hunk 1]
```diff
@@ -29,6 +29,15 @@
 _ARTIFACT_UPLOAD_BATCH_SIZE = (
     50  # Max number of artifacts for which to fetch write credentials at once.
 )
+_AWS_MIN_CHUNK_SIZE = 5 * 1024**2  # 5 MB is the minimum chunk size for S3 multipart uploads
+
+
+def _validate_chunk_size_aws() -> None:
+    chunk_size = MLFLOW_MULTIPART_UPLOAD_CHUNK_SIZE.get()
+    if chunk_size <= _AWS_MIN_CHUNK_SIZE:
```
Comment (harupy) [inline mlflow/store/artifact/cloud_artifact_repo.py:37]:
```suggestion
    if chunk_size < _AWS_MIN_CHUNK_SIZE:
```

to allow `5 * 1024**2`.

Comment (harupy) [inline mlflow/store/artifact/cloud_artifact_repo.py:37]: can we also validate the chunk size doesn't exceed the upper limit? 

https://docs.aws.amazon.com/AmazonS3/latest/userguide/qfacts.html

File: mlflow/store/artifact/cloud_artifact_repo.py [hunk 2]
```diff
@@ -29,6 +29,15 @@
 _ARTIFACT_UPLOAD_BATCH_SIZE = (
     50  # Max number of artifacts for which to fetch write credentials at once.
 )
+_AWS_MIN_CHUNK_SIZE = 5 * 1024**2  # 5 MB is the minimum chunk size for S3 multipart uploads
+
+
+def _validate_chunk_size_aws() -> None:
+    chunk_size = MLFLOW_MULTIPART_UPLOAD_CHUNK_SIZE.get()
```
Comment (harupy) [inline mlflow/store/artifact/cloud_artifact_repo.py:36]:
```suggestion
def _validate_chunk_size_aws(chunk_size: int) -> None:
```

Can we pass chunk size to this function so the code would look like this?


```python
chunk_size = ...
_validate_chunk_size_aws(chunk_size)
```

Comment (harupy) [reply mlflow/store/artifact/cloud_artifact_repo.py:36]:
```suggestion
def _validate_chunk_size_aws() -> int:
```

We can return the passed-in chunk if that's convenient.

File: mlflow/store/artifact/cloud_artifact_repo.py (hunk 1 above)
Comment (ian-ack-db) [reply mlflow/store/artifact/cloud_artifact_repo.py]: Changed

File: mlflow/store/artifact/cloud_artifact_repo.py (hunk 2 above)
Comment (ian-ack-db) [reply mlflow/store/artifact/cloud_artifact_repo.py]: Changed to take the chunk size as a parameter. Kept None output for simpler testing / mocking setups.

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**